cloud/pkg/cloudstream/tunnelserver.go

```diff
@@ -126,7 +125,19 @@ func (s *TunnelServer) connect(r *restful.Request, w *restful.Response) {
 		sessionID:     hostNameOverride,
 	}
 
-	s.updateNodeKubeletEndpoint(hostNameOverride)
+	err = s.updateNodeKubeletEndpoint(hostNameOverride)
+	if err != nil {
+		klog.Errorf("Failed to Update KubeletEndpoint Port, send reconnect msg to edgeStream")
+		for retry := 0; retry < 3; retry++ {
+			msg := stream.NewMessage(0, stream.MessageTypeReconnect, []byte(err.Error()))
+			if err := session.tunnel.WriteMessage(msg); err == nil {
+				klog.V(4).Infof("CloudStream send reconnect message to edge successfully")
+				return
+			}
+			klog.Errorf("CloudStream failed to send reconnect message to edge, error: %v", err)
+		}
+		return
```

Comment:
First of all, does the retry here make sense?
Then, the `reconnect` expression is not appropriate, I think the disconnection type message should be sent directly, similar to `CloseMessage`.

Reply:
The first is to consider, retry if the send close message is unsuccessful。

Reply:
Yes, this practice is common.
But to be sure, is it necessary?
Under what circumstances will send fail?
Even if the sending fails, what will be the impact of not doing other processing?